packages/vc-verification/src/credential-resolver.ts

```diff
@@ -0,0 +1,111 @@
+import { providers } from 'ethers';
+import { CredentialResolverSettings } from './models';
+import { DidStore } from '@ew-did-registry/did-ipfs-store';
+import { IDidStore } from '@ew-did-registry/did-store-interface';
+import { OffChainClaim } from './models';
+import { isValidErc1056, getDIDChain } from '@ew-did-registry/did';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import * as jwt from 'jsonwebtoken';
+
+export class CredentialResolver {
```

Comment:
I think it makes sense to be a bit more specific with the class name as I predict that we will have other types of `CredentialResolvers` in the future.
E.g. I anticipate having a `VuiIssuerApiCredentialResolver` in the future (https://gataca-io.github.io/vui/issuerResolution#resolution-api)
My suggested name might be a bit wordy, but let me know what you think.
```suggestion
export class DidServiceIpfsCredentialResolver {
```

Reply:
The `CredentialResolver` could be the upfront class which manages / instantiates the specific resolvers like `DidServiceIpfsCredentialResolver` and `VuiIssuerApiCredentialResolver`. Thus the underlying approach would be abstracted by `CredentialResolver` and we can have as many resolver approaches as we want. The `CredentialResolverSettings` could handle the settings for different type of resolution options and be provided to `CredentialResolver`. What do you think?

Reply:
I think it is best if the client can pass their own implementation, without needing to modify `CredentialResolver`. It is not good if `CredentialResolver` needs to be modified for every new "type" of credential resolver.
I'm open to having `CredentialResolver` being a factory which instantiates the various types of resolvers. If you like that idea, I would change the name to `CredentialResolverFactory` and have it accept the settings object you describe and have it return `DidServiceIpfsCredentialResolver`. Personally, I don't think it is really necessary to have such a factory at this point. I personally don't like the idea of a single `CredentialResolverSettings` type that is used for both `DidServiceIpfsCredentialResolver` and `VuiIssuerApiCredentialResolver` as it couples the construction of the various types of `CredentialResolver`.

Reply:
Ok, separate implementation of resolver being passed by the client as per their choice and `CredentialResolver` able to verify them sounds right.